tests/test_lab_dev/test_states/test_states_base.py

```diff
@@ -677,3 +677,13 @@ def test_random(self, modes):
         )  # checks if the off-diagonal block of dm is PSD
         assert np.all(np.linalg.eigvals(Gamma) < 1)
         assert np.all(np.linalg.eigvals(Temp) < 1)
+
+    @pytest.mark.parametrize("modes", [[1], [9, 2], [0, 1, 2, 3, 4]])
+    def test_is_positive(self, modes):
+        assert (Ket.random(modes) >> Attenuator(modes)).is_positive
```

Comment:
Can we have these tests for both Ket and DM where each is used respectively? E.g. TestKet -> Ket.random and TestDM -> DM.random

Reply:
I am not exactly sure what you meant, but I think I get it. I changed the tests accordingly. Please let me know if there is still something missing or if I did not address your concern.